cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -624,6 +625,17 @@ def _validate(self, name: str):
                 ),
                 FailureLevel.ERROR,
             )
+        if len(name) > SHARED_STORAGE_NAME_MAX_LENGTH:
```

Comment:
let's add a unit tests for this and for `^default$` case.

Reply:
Added unit tests for SharedStorageNameValidator